pkg/collector/corechecks/snmp/checkconfig/config_validate_enrich.go

```diff
@@ -42,45 +60,81 @@ func validateEnrichMetrics(metrics []MetricsConfig) []string {
 			}
 			for i := range metricConfig.MetricTags {
 				metricTag := &metricConfig.MetricTags[i]
-				errors = append(errors, validateEnrichMetricTag(metricTag, metricConfig)...)
+				errors = append(errors, validateEnrichMetricTag(metricTag)...)
 			}
 		}
 	}
 	return errors
 }
 
-func validateEnrichSymbol(symbol *SymbolConfig, metricConfig *MetricsConfig) []string {
+// validateEnrichMetadata will validate MetadataConfig and enrich it.
+func validateEnrichMetadata(metadata MetadataConfig) []string {
+	var errors []string
+	for resName := range metadata {
+		_, isValidRes := validMetadataResources[resName]
+		if !isValidRes {
+			errors = append(errors, fmt.Sprintf("invalid resource: %s", resName))
+		} else {
+			res := metadata[resName]
+			for fieldName := range metadata[resName].Fields {
```

Comment:
🤏 you might use `res` here:
```suggestion
			for fieldName := range res.Fields {
```